profiles/models.py

```diff
@@ -51,12 +53,48 @@ def __ne__(self, other: Any) -> bool:
         return not self.__eq__(other)
 
 
+class Affiliation(db.Model):
+    # pylint: disable = too-many-instance-attributes
+    id = db.Column(db.Text(), primary_key=True)
+    department = db.Column(db.Text())
+    organisation = db.Column(db.Text(), nullable=False)
+    city = db.Column(db.Text(), nullable=False)
+    region = db.Column(db.Text())
+    country = db.Column(ISO3166Country, nullable=False)
+    starts = db.Column(UTCDateTime, nullable=False)
+    ends = db.Column(UTCDateTime)
+    restricted = db.Column(db.Boolean(), nullable=False)
+    profile_id = db.Column(db.String(ID_LENGTH), db.ForeignKey('profile.id'))
+    profile = db.relationship('Profile', back_populates='affiliations')
+    position = db.Column(db.Integer())
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, affiliation_id: str, country: Country, organisation: str, starts: datetime,
```

Comment:
perhaps an Address object, as the API suggests, could be useful; not sure if it's a nightmare to map onto the database

Reply:
(as a Value Object decomposed into multiple fields on the owning entity, only advanced ORMs support it http://docs.oracle.com/javaee/6/api/javax/persistence/Embeddable.html)

Reply:
Done already for the name: https://github.com/elifesciences/profiles/blob/develop/profiles/models.py#L56-L58

(Not as smooth as it could be.)

Reply:
Not bad, boilerplate is limited to one line per decomposed hidden field (which is needed to decide on the column type anyway) plus the decomposition method

Reply:
Our API does also have an 'address' concept.